asciidoctorj-documentation/src/main/asciidoc/extension-migration-guide.adoc

```diff
@@ -0,0 +1,297 @@
+
+| Document             | DocumentRuby
+| ContentNode          | AbstractNode
+| StructuralNode       | AbstractBlock
+| Block                | Block
+| Section              | Section
+| List                 | List
+| ListItem             | ListItem
+| DescriptionList      | N/A
+| DescriptionListEntry | N/A
+| Table                | Table
+| Column               | Column
+| Row                  | Row
+| Cell                 | Cell
+| PhraseNode           | Inline
+|===
+
+As you can see not all AST classes were renamed, but in particular those classes that appear in the signatures of the processor classes were renamed.
+
+After renaming the classes the new Processor looks like this:
+
+.YellStaticBlock.java after renaming the AST classes
+[source]
+----
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.asciidoctor.ast.StructuralNode;
+import org.asciidoctor.extension.BlockProcessor;
+import org.asciidoctor.extension.Reader;
+
+public class YellStaticBlock extends BlockProcessor {
+
+    private static Map<String, Object> configs = new HashMap<String, Object>() {{
+        put("contexts", Arrays.asList(":paragraph"));
+        put("content_model", ":simple");
+    }};
+
+    public YellStaticBlock(String name, Map<String, Object> config) {
+        super(name, configs);
+    }
+
+    @Override
+    public Object process(StructuralNode parent, Reader reader, Map<String, Object> attributes) {
+        List<String> lines = reader.readLines();
+        String upperLines = null;
+        for (String line : lines) {
+            if (upperLines == null) {
+                upperLines = line.toUpperCase();
+            }
+            else {
+                upperLines = upperLines + "\n" + line.toUpperCase();
+            }
+        }
+
+        return createBlock(parent,
+            "paragraph",
+            Arrays.asList(upperLines),
+            attributes,
+            new HashMap<Object, Object>());
+    }
+}
+----
+
+Together with the AST class names also the factory methods of the common interface of all extensions, `org.asciidoctor.extension.Processor` were renamed.
+While this isn't a problem here, for example invocations of `createInline()` have to be renamed to `createPhraseNode()` according to the <<table-ast-class-names,table above>>.
+
+This extension will already run with AsciidoctorJ 1.6.0 and the following test will pass:
+
+[source,indent="0"]
+----
+        Asciidoctor asciidoctor = Asciidoctor.Factory.create();
+        asciidoctor.javaExtensionRegistry().block("yell", YellStaticBlock.class);
+
+        final String doc = "[yell]\nHello World";
+
+        final String result = asciidoctor.convert(doc, OptionsBuilder.options());
+        Document htmlDoc = Jsoup.parse(result);
+        assertEquals("HELLO WORLD", htmlDoc.select("p").first().text());
+----
+
+Now there's a few things you can do to further make the extension more concise.
```

Comment:
shoudn't it be "there're"?

Reply:
"there're" isn't a valid contraction. But you are correct that "there's" isn't appropriate here. I would write this as:

> There are some additional steps you can take to make this extension more concise.